src/pages/collaboration/CollaborationRegister.tsx

```diff
@@ -0,0 +1,384 @@
+    const file = event.target.files?.[0];
+    if (file && file.type === 'application/pdf') {
+      setPdfFile(file);
+    }
+  };
+
+  const handlePdfClick = () => {
+    pdfInputRef.current?.click();
+  };
+
+  const handlePriceChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replace(/[^0-9]/g, '');
+    setPrice(value);
+  };
+
+  const handleQuantityChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value.replace(/[^0-9]/g, '');
+    const numberValue = value === '' ? 0 : parseInt(value, 10);
+    setQuantity(numberValue);
+  };
+
+  const handleIncrement = () => {
+    setQuantity((prev) => prev + 1);
+  };
+
+  const handleDecrement = () => {
+    setQuantity((prev) => (prev > 0 ? prev - 1 : 0));
+  };
+
+  const handlePortfolioVisibility = () => {
+    setIsPortfolioVisible((prev) => !prev);
+  };
+
+  const handleEditClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+  };
+
+  const handleCancel = () => {
+    navigate(-1);
+  };
+
+  const handleSubmit = () => {
+    navigate('/idea-market/registered');
+  };
+
+  useEffect(() => {
+    return () => {
+      if (previewImageUrl) {
+        URL.revokeObjectURL(previewImageUrl);
+      }
+    };
+  }, [previewImageUrl]);
+
+  const modules = useMemo(
+    () => ({
+      toolbar: {
+        container: [
+          [{ font: [] }, { size: [] }, { align: [] }],
+          ['link', 'image'],
+        ],
+        handlers: {
+          image: () => {
+            const input = document.createElement('input');
+            input.setAttribute('type', 'file');
+            input.setAttribute('accept', 'image/*');
+            input.click();
+
+            input.onchange = async () => {
+              const file = input.files?.[0];
+              if (file) {
+                const reader = new FileReader();
+                reader.onload = () => {
+                  const quill = quillRef.current?.getEditor();
+                  if (quill) {
+                    const range = quill.getSelection(true);
+                    quill.insertEmbed(range.index, 'image', reader.result);
+                  }
+                };
+                reader.readAsDataURL(file);
+              }
+            };
+          },
+        },
+      },
+    }),
+    [],
+  );
+
+  const formats = ['font', 'size', 'align', 'link', 'image'];
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.title}>팀 빌딩</div>
+      <div className={styles.horizontalContainer}>
+        <div className={`${styles.formGroup} ${styles.categoryGroup}`}>
+          <div className={styles.labelWrapper}>
+            <label htmlFor='category'>
+              주제 분야 설정
+              <span className={styles.required}>(필수)</span>
+            </label>
+          </div>
+          <div
+            className={styles.select}
+            onClick={() => setIsDropdownOpen(!isDropdownOpen)}>
```

Comment:
값을 직접 바꾸면 의도치 않은 결과가 날 수 있기 때문에 피하시는 것이 좋습니다!
```suggestion
            onClick={() => setIsDropdownOpen((prev) => !prev)}>
```